multinet/auth/types.py

```diff
@@ -0,0 +1,54 @@
+"""Types associated with authentication."""
+
+from typing_extensions import TypedDict
+
+
+class GoogleUserInfo(TypedDict):
+    """Representation of raw User Info from a Google Account."""
+
+    family_name: str
+    given_name: str
+    name: str
+    picture: str
+    locale: str
+    sub: str  # Unique identifier
+    email: str
+    email_verified: bool
+
+    # Time of token expiration in Unix Seconds
+    exp: int
+
+    # Time of token issue
+    iat: int
+
+    # Issuer
+    iss: str
+
+    aud: str
+    azp: str
+    at_hash: str
+    hd: str
+
+
+class UserInfo(TypedDict):
+    """The filtered info that is used to create a User."""
+
+    family_name: str
+    given_name: str
+    name: str
+    picture: str
+    email: str
+    sub: str  # Unique identifier
+
+
+class MultinetInfo(TypedDict):
+    """Data specific to multinet."""
+
+    session: str
+
+
+# TODO: Should we include ArangoDB document metadata in this?
```

Comment:
I'd vote no if it can easily be retrieved from the database

Reply:
Why would we include the metadata?

Reply:
Sorry, this is a bit confusing. There's parts of the code where we use the `User` type, but there's actually extra arango document data in there, and the spots where that data is/isn't there is unclear right now. I'll have to address this.

Reply:
Is this a problem for typechecking, or some other concern?

Reply:
This is addressed in 15d5819. Now `User` includes arango metadata (`_id`, `_key`, `_rev`), and `FilteredUser` (which is what's returned by the API) strips them out.